numba/np/arraymath.py

```diff
@@ -3781,20 +3781,41 @@ def searchsorted_impl(a, v, side='left'):
 def np_digitize(x, bins, right=False):
     @register_jitable
     def are_bins_increasing(bins):
-        n = len(bins)
-        is_increasing = True
-        is_decreasing = True
-        if n > 1:
-            prev = bins[0]
-            for i in range(1, n):
-                cur = bins[i]
-                is_increasing = is_increasing and not prev > cur
-                is_decreasing = is_decreasing and not prev < cur
-                if not is_increasing and not is_decreasing:
-                    raise ValueError("bins must be monotonically increasing "
-                                     "or decreasing")
-                prev = cur
-        return is_increasing
+
+        # all bin edges hold the same value
+        if len(bins) == 0:
+            return True
+
+        # Skip repeated values at the beginning of the array
+        last_value = bins[0]
```

Comment:
Sure thing - I replicated the corresponding numpy C algorithm relatively closely in an effort to replicate its behaviour, including the name choices (albeit suffixed to avoid shadowing the built-in `next`) but there's no strict requirement to do so.